src/commons/withScrollReached.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {useState, useCallback} from 'react';
+import {
+  FlatListProps,
+  ScrollViewProps,
+  NativeSyntheticEvent,
+  NativeScrollEvent
+} from 'react-native';
+// eslint-disable-next-line no-unused-vars
+import forwardRef, {ForwardRefInjectedProps} from './forwardRef';
+
+export type ScrollReachedProps = {
+  onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void;
+  isScrollAtStart?: boolean;
+  isScrollAtEnd?: boolean;
+};
+
+type SupportedViews = FlatListProps<any> | ScrollViewProps;
+
+export type WithScrollReachedProps = SupportedViews & {
+  scrollReachedProps: ScrollReachedProps;
+  ref?: any;
+};
+type PropTypes = ForwardRefInjectedProps & SupportedViews;
```

Comment:
Not sure about the type here? 
Basically you're saying that the Wrapped component must be a ScrollView/FlatList? 
But isn't it possible the Wrapped component is just a simple View that wraps ScrollView/FlatList ?

Reply:
I think that what matters is this:
`WrappedComponent: React.ComponentType<WithScrollReachedProps>`
maybe I should rename this to something else?
BTW, this section is changed because of other warnings\errors, but please let me know if the name (or my thinking) is wrong.

Reply:
it's not about the name. 

I think there's might be something wrong with the types, but maybe i'm missing something, let's take it offline